code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm

```diff
@@ -3,20 +3,107 @@
 	acid_type = /obj/effect/xenomorph/acid/weak
 
-/datum/action/xeno_action/activable/salvage_plasma
-	name = "Salvage Plasma"
+/datum/action/xeno_action/activable/salvage_biomass
+	name = "Salvage Biomass"
 	action_icon_state = "salvage_plasma"
-	ability_name = "salvage plasma"
-	var/plasma_salvage_amount = PLASMA_SALVAGE_AMOUNT
-	var/salvage_delay = 3 SECONDS
-	var/max_range = 1
+	ability_name = "salvage biomass"
 	keybind_signal = COMSIG_XENOABILITY_SALVAGE_PLASMA
+	cooldown_timer = DRONE_SALVAGE_COOLDOWN
+
+/datum/action/xeno_action/activable/salvage_biomass/on_cooldown_finish()
+	owner.playsound_local(owner, 'sound/voice/alien_drool1.ogg', 25, 0, 1)
+	to_chat(owner, "<span class='notice'>We are ready to salvage xeno biomass again.</span>")
+	return ..()
+
+/datum/action/xeno_action/activable/salvage_biomass/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+
+	if(QDELETED(A))
+		return FALSE
 
-/datum/action/xeno_action/activable/salvage_plasma/use_ability(atom/A)
 	var/mob/living/carbon/xenomorph/X = owner
-	if(owner.action_busy)
-		return
-	X.xeno_salvage_plasma(A, plasma_salvage_amount, salvage_delay, max_range)
+	var/mob/living/carbon/xenomorph/target = A
+
+	if(!isxeno(target))
+		to_chat(X, "<span class='xenowarning'>We can only salvage the biomass of deceased xenomorphs!</span>")
+		return FALSE
+
+	if(target.stat != DEAD)
+		to_chat(X, "<span class='xenowarning'>We can't salvage the biomass of living sisters!</span>")
+		return FALSE
+
+	var/distance = get_dist(X, target)
+	if(distance > DRONE_SALVAGE_BIOMASS_RANGE)
+		to_chat(X, "<span class='xenowarning'>We need to be [distance -  DRONE_SALVAGE_BIOMASS_RANGE] tiles closer to [target].</span>")
+		return FALSE
+
+
+/datum/action/xeno_action/activable/salvage_biomass/use_ability(atom/A)
+
+	var/mob/living/carbon/xenomorph/X = owner
+	var/mob/living/carbon/xenomorph/target = A
+
+
+	X.face_atom(target) //Face the target so we don't look like an ass
+	to_chat(X, "<span class='xenowarning'>We being to salvage the biomass of [target]...</span>")
+	playsound(X, pick('sound/voice/alien_drool1.ogg','sound/voice/alien_drool2.ogg'), 25)
+
+	if(!do_after(X, DRONE_SALVAGE_BIOMASS_WINDUP, TRUE, target, BUSY_ICON_FRIENDLY)) //Wind up
+		return fail_activate()
+
+
+	X.face_atom(target) //Face the target so we don't look like an ass
+
+	if(!can_use_ability(target, FALSE, XACT_IGNORE_PLASMA)) //Check to make sure the target hasn't gone anywhere/been exploded/etc
+		return fail_activate()
+
+	succeed_activate()
+
+	var/plasma_amount = target.plasma_stored //We take all the target's plasma; they won't need it where they're going/are
+	var/upgrade_amount = target.upgrade_stored * DRONE_SALVAGE_BIOMASS_SALVAGE_RATIO //We only recover a small portion of the target's upgrade and evo points.
+	var/evo_amount = target.evolution_stored * DRONE_SALVAGE_BIOMASS_SALVAGE_RATIO
+
+	//Take all the plas-mar
+	target.use_plasma(plasma_amount)
+	X.gain_plasma(plasma_amount)
```

Comment:
Why would we remove the plasma if we gib them after anyway

Reply:
It's residual code; removed